src/submission-handlers/encounterDatetimeHandler.test.ts

```diff
@@ -0,0 +1,159 @@
+import { type EncounterContext } from '../form-context';
+import { type OpenmrsEncounter, type FormField } from '../types';
+import { EncounterDatetimeHandler } from './encounterDatetimeHandler';
+
+const encounterContext: EncounterContext = {
+  patient: {
+    id: '833eb896-c1f0-11eb-8529-0242ac130003',
+  },
+  location: {
+    uuid: '81e6e516-c1f0-11eb-8529-0242ac130003',
+  },
+  encounter: {
+    uuid: '773455da-3ec4-453c-b565-7c1fe35426be',
+    encounterDatetime: '2023-06-09T09:42:40+00:00',
+    encounterProviders: [],
+    obs: [],
+  },
+  sessionMode: 'enter',
+  encounterDate: new Date(2020, 11, 29),
+  setEncounterDate: (value) => {},
+  encounterProvider: '2c95f6f5-788e-4e73-9079-5626911231fa',
+  setEncounterProvider: jest.fn,
+  setEncounterLocation: jest.fn,
+  encounterRole: '6d95f6f5-788e-4e73-9079-5626911231f4',
+  setEncounterRole: jest.fn,
+};
+
+describe('EncounterDatetimeHandler', () => {
+  let field: FormField;
+
+  beforeEach(() => {
+    // Define the field once before each test
+    field = {
+      label: 'Encounter Location',
+      type: 'encounterLocation',
+      required: false,
+      id: 'encounterLocation',
+      questionOptions: {
+        rendering: 'ui-select-extended',
+      },
+      validators: [],
+    };
+  });
+
+  afterEach(() => {
+    // Clean up any side effects if needed
+    jest.clearAllMocks();
+  });
+
+  describe('handleFieldSubmission', () => {
+    it('should handle encounter date time submission', () => {
+      // replay
+      const role = EncounterDatetimeHandler.handleFieldSubmission(field, '2017-01-18T09:42:40+00:00', encounterContext);
+      // verify
+      expect(role).toEqual('2017-01-18T09:42:40+00:00');
+    });
```

Comment:
This test tests nothing. The function `handleFieldSubmission` is two lines long and it obviously returns the value of the second argument.

If you want to make it do something, mock `setEncounterDate` and check that it was called with the correct argument.